usaon_vta_survey/routes/user.py

```diff
@@ -16,7 +16,10 @@
 
 
 def _validate_role_change(user: User, form) -> None:
-    if not form.data['role'] == user.role_id and not current_user.role_id == 'admin':
+    selected_value = form.role().split("option selected value=")
+    selected_value = selected_value[1].split("<")
+    selected_value = selected_value[0].split(">")[1]
+    if not selected_value == user.role_id and not current_user.role_id == 'admin':
```

Comment:
I'm struggling to understand this. Can you give a summary of what the cause of the issue was and how this resolves it? It looks like you're string parsing an HTML tag, which I hope we can avoid entirely!

The conditional in general should be rewritten for readability, e.g.:

```python
can_edit = (
  thing_1 == thing_2
  and thing3 != thing_4
)

if not can_edit:
  raise RuntimeError("...")
```

Reply:
Basically, when I looked through what we were doing before the first condition was always X==X if any field was touched on profile. So this prevented any user besides an `admin` from editing a profile. When I searched through the fields `form.role()` was the only one to show me what is CURRENTLY being selected so I did have to parse through that to compare the selected value with the current value to see if the user was looking to change their role.

Reply:
Can we access `form.data.role` instead of parsing the HTML?

Reply:
We were previously accessing `form.data['role']` and that was not working because on ANY change it was triggering. Not JUST a change to the role field.

Reply:
@MattF-NSIDC even though parsing the HTML is not pretty, I believe it is the best solution at this moment given the time available. Unless there is a security reason why we shouldn't do this I think I'm gonna go ahead and merge.

Reply:
>We were previously accessing form.data['role'] and that was not working because on ANY change it was triggering. Not JUST a change to the role field.

Why was this the case? Was the value you were receiving somehow different from the one we're parsing out of the HTML? I feel like we don't have a clear understanding of the root cause yet.

>Unless there is a security reason why we shouldn't do this I think I'm gonna go ahead and merge.

If you're going to merge I do feel the readability issues warrant at least opening an issue to come back and resolve this, and a comment in the code linking to that issue! There must be something we're missing. We should never have to parse HTML to access user input.

Reply:
Nice! Thank you for digging deeper in to this :)